cjworkbench/tests/i18n/util.py

```diff
@@ -7,15 +7,14 @@
 
 @contextmanager
 def mock_app_catalogs(catalogs: Dict[str, Catalog]):
-    # Code to acquire resource, e.g.:
-    old_localizer = MESSAGE_LOCALIZER_REGISTRY._app_localizer
-    try:
-        new_localizer = MessageLocalizer(catalogs)
-        MESSAGE_LOCALIZER_REGISTRY._app_localizer = new_localizer
-        yield new_localizer
-    finally:
-        # Code to release resource:
-        MESSAGE_LOCALIZER_REGISTRY._app_localizer = old_localizer
+    def mock_for_application(registry):
```

Comment:
You don't need `@contextmanager` to decorate `patch()` -- you can just return from `patch()` and that will be a context manager.

And if you're using `patch()`, then consider `patch.object()` -- it's less finicky. (Your usage here is correct in this case, but `patch.object()` is a better pattern in general.)

I propose:

```
def mock_app_catalogs(catalogs: Dict[str, Catalog]) -> ContextManager:
    return patch.object(MODULE_LOCALIZER_REGISTRY, "application_localizer", MessageLocalizer(catalogs))
```